src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -767,10 +801,11 @@ private static boolean shouldCheckSeparationFromPreviousToken(DetailAST ast) {
      * @return true if it should be checked if next token is separated by whitespace,
      *      false otherwise.
      */
-    private static boolean shouldCheckSeparationFromNextToken(DetailAST ast, char nextChar) {
+    private boolean shouldCheckSeparationFromNextToken(DetailAST ast, char nextChar) {
         return !(ast.getType() == TokenTypes.LITERAL_RETURN
                     && ast.getFirstChild().getType() == TokenTypes.SEMI)
-                && ast.getType() != TokenTypes.ARRAY_INIT
+                && (ast.getType() != TokenTypes.ARRAY_INIT
+                || allowWhitespaceAroundArrayInit)
```

Comment:
@nmancus1 It shows an error over here-
```
(extension) LogicConditionNeedOptimization: Condition with || at line 808 position 16 need optimization. All method calls are advised to move to end of logic expression.
```
It says to improve logic, but I have used the most optimized logic I can.  Using this or statement is very crucial for check to perform in both `true` and `false` state. Any workaround for this problem?

Reply:
@rnveach Can you help, you opened a related issue in past. https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/683

Reply:
Resolved after reading `LogicConditionNeedOptimization`

Reply:
```
 && (ast.getType() != TokenTypes.ARRAY_INIT
                || allowWhitespaceAroundArrayInit)
```
to 
```
&& (allowWhitespaceAroundArrayInit)
|| ast.getType() != TokenTypes.ARRAY_INIT)

```

methods should be the last statements possible.